hcipy/aperture/realistic.py

```diff
@@ -750,3 +750,210 @@ def segment_with_spider(segment):
+		if with_spiders:
+			
+			spider_attachement_mask = 1 - regular_polygon_aperture(3, truss_size, angle=np.pi, center=None)(grid) 
+
+			spider_mask = grid.ones()	
+			for i in range(3):
+				offset_angle = 2 * np.pi / 3 * i
+				spider_1_start = radius_spider_1 * np.array([np.sin(offset_angle), np.cos(offset_angle)])
+				spider_mask *= 1 - make_diverging_spider(spider_1_start, spider_1_width, np.deg2rad(1.16), np.deg2rad(9.83) + offset_angle)(grid)
+				spider_mask *= 1 - make_diverging_spider(spider_1_start, spider_1_width, np.deg2rad(1.16), np.deg2rad(180.0 - 9.83) + offset_angle)(grid)
+				
+				spider_2_start = radius_spider_2 * np.array([np.sin(offset_angle), np.cos(offset_angle)]) + offset_spider_2 * np.array([np.cos(offset_angle), -np.sin(offset_angle)]) 
+				spider_mask *= 1 - make_diverging_spider(spider_2_start, spider_2_width, np.deg2rad(0.0), np.deg2rad(-11.2) + offset_angle)(grid)
+
+			return center_segment * spider_mask * spider_attachement_mask
+
+		else:
+			return center_segment
+
+	segment_functions = [make_central_gmt_segment,]
+	for i in range(6):
+		rotation_angle = np.pi/3 * i
+		xc = segment_distance * np.cos(rotation_angle)
+		yc = segment_distance * np.sin(rotation_angle)
+		
+		aperture = make_shifted_aperture( make_rotated_aperture(elliptical_aperture([off_axis_segment_size * np.cos(off_axis_tilt), off_axis_segment_size]), rotation_angle), [xc, yc])
+		segment_functions.append(aperture)
+
+	# Center segment obscurations
+	def make_aperture(grid):
+		aperture = grid.zeros()
+		for segment in segment_functions:
+			aperture += segment(grid)
+		return aperture
+
+	if return_segments:
+		return make_aperture, segment_functions
+	else:
+		return make_aperture
+
+def make_tmt_aperture(normalized=False, with_spiders=True, return_segments=False):
+	'''Make the Thirty-Meter Telescope aperture.
+
+	The aperture is based on the description from https://www.tmt.org/page/optics. The size of
+	the secondary and the spiders were derived from Figure 5 of [JensenClem2021]_
+
+	.. [JensenClem2021] Rebecca Jensen-Clem et al. "The Planetary Systems Imager Adaptive Optics System:
+	 An Initial Optical Design and Performance Analysis Tools for the PSI-Red AO System" SPIE Vol. 11823 (2021)
+
+	Parameters
+	----------
+	normalized : boolean
+		If this is True, the outer diameter will be scaled to 1. Otherwise, the
+		diameter of the pupil will be 30.0 meters.
+	with_spiders : boolean
+		If this is False, the spiders will be left out. Default: True.
+	return_segments : boolean
+		If this is True, the segments will also be returned as a list of Field generators.
+
+	Returns
+	-------
+	Field generator
+		The TMT aperture.
+	tmt_segments : list of Field generators
+		The segments. Only returned when `return_segments` is True.
+	'''
+
+	tmt_outer_diameter = 30.0
+	spider_width = 0.22
+	segment_size = 1.44
+	segment_gap = 0.0025
+	inner_diameter = 2.5 * segment_size
+	outer_diameter = 30.0
```

Comment:
Why the duplicate outer diameter parameter?

Reply:
Removed duplicate.